game/script-farewells.rpy

```diff
@@ -39,38 +49,377 @@ init python in jn_farewells:
+        extend 1uchgn " See you later,{w=0.1} [player]!"
+
+    elif jn_get_holiday_for_date() == JNHolidays.halloween:
+        n 1usqss "Ooh?{w=0.2} Heading out for Halloween,{w=0.1} [player]?"
+        n 1fsqsm "Just don't forget..."
+        n 1fsqbg "I want my share of treats too!"
+        n 1fchgn "Ehehe.{w=0.5}{nw}"
+        extend 1uchbg " Have fun~!"
+
+    elif jn_get_holiday_for_date() == JNHolidays.christmas_eve:
+        n 1unmbo "Oh?{w=0.2} You're heading out for Christmas Eve?"
+        n 1kllsl "Well...{w=0.3} okay."
+        n 1kllajl "...You will be back in time for Christmas though...{w=0.5}{nw}" 
+        extend 1knmsll " right?"
+        n 1klrbgl "...Ahaha.{w=0.3}" 
+        extend 1kchbg " See you later,{w=0.1} [player]!"
+
+    elif jn_get_holiday_for_date() == JNHolidays.christmas_day:
+        n 1unmbo "Huh?{w=0.2} You're heading off now?"
+        n 1kllsl "Well...{w=0.3} alright."
+        n 1kllss "Thanks for dropping by today though,{w=0.1} [player]."
+        n 1kcsssl "It...{w=0.3} really meant a lot to me."
+        n 1kchss "See you later,{w=0.1} [player]!{w=0.5}{nw}" 
+        extend 1kchbg " And Merry Christmas!"
+
+    else:
+        n 1unmaj "Oh?{w=0.2} You're heading out,{w=0.1} [player]?"
+        n 1fchbg "No worries!{w=0.2} I'll see you later then,{w=0.1} 'kay?"
+        n 1nchbg "Bye-{w=0.1}bye,{w=0.1} [player]!"
+
+    if jn_affinity.get_affinity_state() >= jn_affinity.LOVE:
+        n 1uchbgf "Love you~!"
+
+    return { "quit": None }
+
+label farewell_option_work:
+    if jn_utils.get_current_hour() >= 20 or jn_utils.get_current_hour() <= 4:
+        n 1knmaj "H-{w=0.1}huh?{w=0.2} You're going to work now?"
+        $ time_concern = "late" if jn_utils.get_current_hour() >= 20 else "early"
+        n 1kllaj "But...{w=0.3} it's super [time_concern] though,{w=0.1} [player]..."
+        n 1kllun "..."
+        n 1fnmun "Just...{w=0.3} be careful,{w=0.1} alright?"
+        extend 1fsqpo " And you {i}better{/i} come visit when you get back."
+        n 1fllsm "...Ehehe."
+        n 1fchbg "Do your best,{w=0.1} [player]!"
+
+    else:
+        n 1unmaj "Oh?{w=0.2} You're working today?"
+        $ holiday_check = jn_get_holiday_for_date()
+
+        if holiday_check == JNHolidays.easter:
+            n 1uskgs "...And on Easter,{w=0.1} of all days?{w=0.5}{nw}" 
+            extend 1fslpo " Man..."
+
+        elif holiday_check == JNHolidays.christmas_eve:
+            n 1fskgsl "...On Christmas Eve?{w=0.5}{nw}"
+            extend 1fcseml " You've gotta be kidding me..."
+
+        elif holiday_check == JNHolidays.christmas_day:
+            n 1fskwrl "...On {i}Christmas{/i}?!{w=0.5}{nw}"
+            extend 1fcseml " Ugh..."
+            n 1fslpol "..."
+            n 1fslajl "Well..."
+
+        elif holiday_check == JNHolidays.new_years_eve:
+            n 1fskgsl "...And on New Year's Eve,{w=0.1} too?!{w=0.5}{nw}"
+            extend 1fcseml " Jeez..."
+
+        elif not jn_utils.get_is_weekday():
```

Comment:
Might be a good idea to name the function `jn_utils.is_weekday()` since it's less a get and more a check, plus makes sense when reading it in code

Reply:
Marking unresolved, at least partially. Global is fine for this, but definitely want to remove `get` from the name for easier readability